### PR TITLE
add some tests

### PR DESCRIPTION
**Pull request by xcaptain**
Still work in progress, test cases are copied from `Casbin.NET` project, in the future more tests should be added.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+   // Use IntelliSense to find out which attributes exist for C# debugging
```
Comment (hsluoyz) [inline .vscode/launch.json]: Please do not commit your .vscode config files.

Comment (xcaptain) [reply .vscode/launch.json]: Thanks, removed.

**Comment by xcaptain** [comment]
@hsluoyz do you think we should implement all the test cases from Casbin.NET or just some?

**Comment by hsluoyz** [comment]
If your adapter supports AutoSave, then you should follow: https://github.com/casbin/xorm-adapter/blob/master/adapter_test.go. If not, then follow: https://github.com/casbin/cassandra-adapter/blob/master/adapter_test.go

I don't think moving into Casbin is a good idea. Because all adapters have their own repo.

And for the commits in this PR, please merge them into one single commit, for easier review.

**Comment by xcaptain** [comment]
@hsluoyz I think this adapter is using autosave and I can't think of any reason to disable autosave. By default `DbContext` in an asp dotnet core project is injected at scoped service, this adapter depends on `DbContext`, so its lifetime must be scoped or transient, which means this adapter will be created per requests, which means write or read from database per request.

Thanks for pointing out how to test an adapter, I will follow your xorm-adapter project.

**Comment by xcaptain** [comment]
@hsluoyz Please take a look, I follow `testAutoSave` in xorm-adapter, the test cases passed in my laptop

**Comment by hsluoyz** [comment]
lgtm. Please squash the commits.

**Comment by xcaptain** [comment]
rebased

**Comment by hsluoyz** [comment]
Merged.

**Comment by hsluoyz** [comment]
Hi @xcaptain , next step, please add the CI and code coverage badges.

**Pull request merged**